hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/HiveHoodieReaderContext.java

```diff
@@ -232,6 +232,12 @@ public UnaryOperator<ArrayWritable> projectRecord(Schema from, Schema to, Map<St
     return HoodieArrayWritableAvroUtils.projectRecord(from, to);
   }
 
+  @Override
+  public Comparable castValue(Comparable value, Schema.Type newType) {
+    //TODO: [HUDI-8261] actually do casting here
```

Comment:
What's preventing from casting here?

Reply:
I just didn't think it was p0. We don't have schema evolution support for hive anyways, so I made a ticket to fix those both in the future. It might not take that long to implement, but the testing infra I would need to create to validate it _**properly**_ might take a couple days.